shard-operator/src/main/java/com/redhat/service/smartevents/shard/operator/ObservabilitySetupService.java

```diff
@@ -0,0 +1,110 @@
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+
+import org.eclipse.microprofile.config.inject.ConfigProperty;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.redhat.service.smartevents.shard.operator.providers.TemplateProvider;
+
+import io.fabric8.kubernetes.api.model.Secret;
+import io.fabric8.kubernetes.client.KubernetesClient;
+import io.quarkus.runtime.StartupEvent;
+
+/**
+ * Initialises Secret required by the Observability Operator in the Data Plane.
+ * Changes to the underlying {@link ConfigProperty} values are handled by the CPaaS
+ * AddOn deployment process and {@link SecretRestartHandler}.
+ */
+@ApplicationScoped
+public class ObservabilitySetupService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(ObservabilitySetupService.class);
+
+    protected static final String OBSERVABILITY_ACCESS_TOKEN = "access_token";
+    protected static final String OBSERVABILITY_REPOSITORY = "repository";
+    protected static final String OBSERVABILITY_CHANNEL = "channel";
+    protected static final String OBSERVABILITY_TAG = "tag";
+    protected static final String OBSERVABILITY_NAME = "smartevents-observability-configuration";
+    protected static final String OBSERVABILITY_NAMESPACE = "smartevents-observability";
+
+    private static final Base64.Encoder ENCODER = Base64.getEncoder();
+
+    @ConfigProperty(name = "event-bridge.observability.access_token")
+    protected Optional<String> optAccessToken;
+    String accessToken;
+
+    @ConfigProperty(name = "event-bridge.observability.repository")
+    protected Optional<String> optRepository;
+    String repository;
+
+    @ConfigProperty(name = "event-bridge.observability.channel")
+    protected Optional<String> optChannel;
+
+    @ConfigProperty(name = "event-bridge.observability.tag")
+    protected Optional<String> optTag;
+
+    @Inject
+    KubernetesClient kubernetesClient;
+
+    @Inject
+    TemplateProvider templateProvider;
+
+    void setupObservabilityComponent(@Observes StartupEvent event) {
+        createOrUpdateObservabilitySecret();
+    }
+
+    protected void createOrUpdateObservabilitySecret() {
+        extractConfigProperties();
+        if (!isObservabilityEnabled()) {
+            return;
+        }
+
+        Secret existing = kubernetesClient
+                .secrets()
+                .inNamespace(OBSERVABILITY_NAMESPACE)
+                .withName(OBSERVABILITY_NAME)
+                .get();
+
+        Secret expected = templateProvider.loadObservabilitySecretTemplate();
+        expected.getData().put(OBSERVABILITY_ACCESS_TOKEN,
+                ENCODER.encodeToString(accessToken.getBytes(StandardCharsets.UTF_8)));
+        expected.getData().put(OBSERVABILITY_REPOSITORY,
+                ENCODER.encodeToString(repository.getBytes(StandardCharsets.UTF_8)));
+        optChannel.ifPresent(channel -> expected.getData().put(OBSERVABILITY_CHANNEL,
```

Comment:
@manstis If we get to this point, don't these values have to be present. Otherwise wouldn't we create a `Secret` with a blank `channel` and `tag`, which I guess ends up in a broken configuration?

It feels like to me that if observability is enabled, then all the other values have to be present otherwise we cannot proceed.

As a suggestion. `@Inject` the `event-bridge.observability.enabled` parameter and also the `org.eclipse.microprofile.config.ConfigProvider`. If `event-bridge.observability.enabled` is `true`, then call the `ConfigProvider.getConfig()` method for all other parameters, failing if any of them are not set.

See: https://quarkus.io/guides/config#programmatically-access-the-configuration

Reply:
OK. Refactored.